modin/pandas/utils.py

```diff
@@ -252,6 +255,88 @@ def check_both_not_none(option1, option2):
     return not (option1 is None or option2 is None)
 
 
+def broadcast_item(
+    query_compiler, row_lookup, col_lookup, item, to_shape=None, need_reindex=True
+):
+    """
+    Use NumPy to broadcast or reshape item.
+
+    Parameters
+    ----------
+    query_compiler : BaseQueryCompiler
+        Query compiler ontains the necessary information about the axes.
```

Comment:
not sure whether we should use QC here as it's not a private method anymore...
DataFrame/Series would be more suitable I think

Reply:
In this case, we have to construct these objects from `query_compiler` just to call `broadcast_item` function, which is not very convenient.

Reply:
The valid point from @dchigarev. We shouldn't use QC at this layer as a parameter. I wonder why `broadcast_item` is a free function now?

Reply:
`broadcast_item` used from `_LocationIndexerBase` and `Dataframe` objects now. I just didn't think of a better option than to make it just a function. I'm open for another solutions :)

Reply:
I see two options here.

1. We can leave `broadcast_item` as a free function, but replace `query_compiler` parameter to any of a high layer object (DataFrame, Indexer, etc.) like here. https://github.com/modin-project/modin/blob/35ae83481aed58c8a8abfc97b900c494a5d47f21/modin/distributed/dataframe/pandas/partitions.py#L28
2. We can split `broadcast_item` into two methods for `DataFrame` and `Indexer`.

Do you think which option would be the best?

Reply:
First option is better here I suppose.

I use `obj` as the name of the variable, just like it is done in functions around.